File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {

    int size = arr.size();
    int x = 1, add;

    while (size > x) {
        x *= 2;
    }

    add = x - size;

    for (int i = 0; i < add; i++) {
        arr.push_back(0);
    }

    return arr;
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.swift
import Foundation

func solution(_ arr:[Int]) -> [Int] {
    
    var size: Double = 0
    var result = arr
    
    while arr.count > Int(pow(2.0, size)) {
        size += 1
    }
    
    var add = Int(pow(2.0, size)) - arr.count
    
    for _ in 0..<add {
        result.append(0)
    }
    
    
    return result
}
